---
title: "Data Visualization"
author: "Mia Isaacs"
date: "2025-02-24"
output: github_document
---

### load packages

```{r}
library(tidyverse)
library(Biostrings)
library(Pviz)
library(Gviz)
library(stringr)
library(GenomicRanges)
```

### view dataset

```{r}
dengueIgG_all = read_csv("data/Dengue_data_analysis_IgG.csv") |> 
  janitor::clean_names()

view(dengueIgG_all)
```


### format as long dataset

```{r}
denv_extract <- dengueIgG_all |> 
  mutate(start_position = as.numeric(str_extract(annotation, "(?<=\\[location=)\\d+")),
         end_position = as.numeric(str_extract(annotation, "(?<=\\.{2})\\d+")))

denv_long <- denv_extract |> 
  pivot_longer(cols = matches("uwa_"),
               names_to = "sample",
               values_to = "fluorescence")

head(denv_long, 10)
colnames(denv_long)
```

### limit dataset to necessary variables

```{r}
denv_clean <- denv_long |> 
  select(annotation, probe_sequence, accession, start_position, end_position, sample, fluorescence)

head(denv_clean, 10)
tail(denv_clean, 10)
```

### separate into travelers vs. ctrl and timepoints
```{r, acute}
all_acute <- denv_clean |> 
  filter(grepl("acute", sample, ignore.case = TRUE)) |> 
  mutate(observation = row_number())
  
view(all_acute)
```


```{r, convalescent T1}
conv_t1 <- denv_clean |> 
  filter(grepl("_1_", sample, ignore.case = TRUE) & grepl("conv", sample, ignore.case = TRUE))

view(conv_t1)
```

```{r, convalescent T2}
conv_t2 <- denv_clean |> 
  filter(grepl("_2_", sample, ignore.case = TRUE))
  
view(conv_t2)
```

```{r, convalescent T3}
conv_t3 <- denv_clean |> 
  filter(grepl("_3_", sample, ignore.case = TRUE))
  
view(conv_t3)
```

```{r, convalescent T4}
conv_t4 <- denv_clean |> 
  filter(grepl("_4_", sample, ignore.case = TRUE))
  
view(conv_t4)
```

```{r, convalescent T5}
conv_t5 <- denv_clean |> 
  filter(grepl("_5_", sample, ignore.case = TRUE))
  
view(conv_t5)
```

```{r, negative control}
neg_ctrl <- denv_clean |> 
  filter(grepl("nc", sample, ignore.case = TRUE))
  
view(neg_ctrl)
```

### import start and end positions
```{r}
positions_df <- read.csv("data/proteome_positions.csv") |> 
  janitor::clean_names()

view(positions_df)
```

### need to figure out how to get the plots i want!! going to make a separate plot for each of the groups above and then i will produce mixed effects models (one general, one for each group)


## mapping humoral response across genome (testing using acute samples)

### convert data frames to tracks
```{r, acute}
options(ucscChromosomeNames=FALSE)

features_gr1 <- GRanges(seqnames = "DENV",
                       ranges = IRanges(start = positions_df$minimum, end = positions_df$maximum),
                       feature = positions_df$name)

probes_gr1 <- GRanges(seqnames = "DENV",
                     ranges = IRanges(start = all_acute$start_position, end = all_acute$end_position),
                     score = log10(all_acute$fluorescence + 1))

genomic_track1 <- GeneRegionTrack(features_gr,
                                 genome = "DENV",
                                 chromosome = "DENV",
                                 name = "Genomic Features",
                                 col = "black",
                                 fill = "lightblue",
                                 showId = TRUE)
probe_track1 <- DataTrack(probes_gr,
                         genome = "DENV",
                         chromosome = "DENV",
                         name = "Reactivity",
                         type = "p",
                         col = "red",
                         ylim = c(0, max(log10(all_acute$fluorescence + 1))))
```

### create dual x-axis
```{r, acute}
genome_axis1 <- GenomeAxisTrack()

probe_axis1 <- GenomeAxisTrack(col = "red")
```

### plot all tracks together
```{r, acute}
plotTracks(list(genome_axis1, genomic_track1, probe_track1, probe_axis1),
           from = min(all_acute$start_position), to = max(all_acute$end_position),
           sizes = c(1, 3, 3, 1))
```


```{r}
summary(all_acute$start_position)
summary(all_acute$end_position)
```




```{r}
zikaIgG_all = read_csv("data/ZIKV_peptides_IgG_peptide_frequency.csv") |> 
  janitor::clean_names()

view(zikaIgG_all)
```